core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -209,9 +209,22 @@ object Battle {
     }
 
     private fun handleCityDefeated(defender: ICombatant, attacker: ICombatant) {
-        if (!defender.isDefeated() || defender !is CityCombatant || attacker !is MapUnitCombatant || !attacker.isMelee()
+        if (!defender.isDefeated() || defender !is CityCombatant || attacker !is MapUnitCombatant || 
+            (!attacker.isMelee() && !attacker.unit.hasUnique(UniqueType.CanDestroyCities, checkCivInfoUniques = true))
             || attacker.unit.hasUnique(UniqueType.CannotCaptureCities, checkCivInfoUniques = true)
         ) return
+
+        // Must come before normal conquest logic so units that cannot capture cities can still destroy them
+        // Melee units can capture capitals; any unit with CanDestroyCities can destroy non-capital cities
+        if (defender.isDefeated() && defender is CityCombatant && attacker is MapUnitCombatant) {
+            if (!defender.city.isCapital()) {
```

Comment:
should be "if (!defender.city.canBeDestroyed()) -> return"
And then the rest of the logic for destruction